server/src/test/java/io/crate/execution/engine/indexing/UpsertResultsTest.java

```diff
@@ -105,4 +111,25 @@ public void test_error_messages_are_limited_to_25() throws Exception {
         Result result = upsertResults.getResultSafe("dummyUri");
         assertThat(result.errors).hasSize(25);
     }
+
+    @Test
+    public void test_summary_collector_can_handle_null_failures() throws Exception {
+        DiscoveryNode node1 = new DiscoveryNode("n1", ESTestCase.buildNewFakeTransportAddress(), Version.CURRENT);
+        UpsertResultCollector summaryCollector = UpsertResultCollectors.newSummaryCollector(
+            node1,
+            AccessControl.DISABLED
+        );
+        UpsertResults upsertResults = new UpsertResults();
+        ArrayList<RowSourceInfo> sourceInfos = new ArrayList<>();
+        ShardResponse shardResponse = new ShardResponse();
+        IllegalStateException dummyError = new IllegalStateException("dummy");
+        for (int i = 0; i < 15; i++) {
```

Comment:
Maybe helps to understand why so much responses are created? 
```suggestion
        // Errors are ignored and null value is used once the failure count is > BULK_RESPONSE_MAX_ERRORS_PER_SHARD
        for (int i = 0; i <= ShardRespons.BULK_RESPONSE_MAX_ERRORS_PER_SHARD + 1; i++) {
```